ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -342,6 +356,240 @@ objects:
+
+          for machine in ${MASTER_MACHINES_TO_REMOVE[@]}
+          do
+            retry 10 oc --request-timeout=5s -n openshift-machine-api delete machine ${machine}
+          done
+
+          echo "Confirm meltdown"
+          sleep 30
+          oc --request-timeout=5s get nodes && exit 1
+
+          echo "Upload a script to restore single node cluster"
+          cat > /tmp/install_recovery_scripts.sh <<'SCRIPT_EOF'
+        #!/bin/bash
+        set -x
+        if [[ $EUID -ne 0 ]]; then
+          echo "This script must be run as root"
+          exit 1
+        fi
+
+        PREFIX="https://raw.githubusercontent.com/vrutkovs/openshift-recovery/unify-fixes"
+
+        curl -s $PREFIX/bin/etcd-member-recover.sh -o /usr/local/bin/etcd-member-recover.sh
+        chmod 755 /usr/local/bin/etcd-member-recover.sh
+
+        curl -s $PREFIX/bin/etcd-snapshot-restore.sh -o /usr/local/bin/etcd-snapshot-restore.sh
+        chmod 755 /usr/local/bin/etcd-snapshot-restore.sh
+
+        curl -s $PREFIX/bin/openshift-recovery-tools -o /usr/local/bin/recovery-tools
+
+        curl -s $PREFIX/bin/tokenize-signer.sh -o /usr/local/bin/tokenize-signer.sh
+        chmod 755 /usr/local/bin/tokenize-signer.sh
+        SCRIPT_EOF
+          chmod +x /tmp/install_recovery_scripts.sh
+
+          retry 10 \
+            scp -o StrictHostKeyChecking=no -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no \
+                -o ServerAliveInterval=30 -W %h:%p core@${BASTION_HOST}" \
+                /tmp/install_recovery_scripts.sh core@"${SURVIVING_MASTER_NODE}":/tmp
+          echo "Run recovery scripts installer"
+          bastion_ssh core@${SURVIVING_MASTER_NODE} "sudo -i /tmp/install_recovery_scripts.sh"
+
+          echo "Restore single etcd"
+          bastion_ssh core@${SURVIVING_MASTER_NODE} "sudo -i /usr/local/bin/etcd-snapshot-restore.sh"
+
+          echo "Wait for API server to come up"
+          retry 30 oc get nodes
+
+          # Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1707006
+          echo "Restart SDN"
+          retry 10 oc delete pods -l app=sdn -n openshift-sdn --wait=false
+
+          echo "Create two masters via Machine API"
+          retry 10 oc get machines -n openshift-machine-api
+          # Clone existing masters, update IDs and oc apply
+          SURVIVING_MASTER_MACHINE=$(oc get machine -l machine.openshift.io/cluster-api-machine-role=master -n openshift-machine-api -o name | cut -d '/' -f 2)
+          SURVIVING_MASTER_NUM=${SURVIVING_MASTER_MACHINE##*-}
+          SURVIVING_MASTER_PREFIX=${SURVIVING_MASTER_MACHINE%-*}
+          oc get --export machine ${SURVIVING_MASTER_MACHINE} -n openshift-machine-api -o yaml > /tmp/machine.template
+          SURVIVING_MASTER_AZ=$(grep -oP 'availabilityZone: (\K.+)' /tmp/machine.template)
+          SURVIVING_MASTER_AZ_INDEX=$(grep -oP 'availabilityZone: .*(\K.)' /tmp/machine.template)
+
+          MASTER_INDEX=0
+          AZ_LIST=("a" "b" "c")
```

Comment:
This seems very brittle (e.g. we currently only use [two zones][1]).  Can we pull it from `oc get -o 'jsonpath={...}' nodes` output or some such instead of hard-coding it?  Or is this enumeration not really about availability zones?

[1]: https://github.com/openshift/release/blob/6ca4a9371305b26407c90c2fe82a98c83b93591a/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L426-L436

Reply:
Fixed in d50d50e33 - a simple loop is used to create two missing masters. Both would be put in the same AZ as the master.

Previously I was attempting to put new masters in other AZs. AZ list had to be kept in sync with install settings, so I dropped it.